src/characters/cbd-recipient.ts

```diff
@@ -46,71 +41,44 @@ export class ThresholdDecrypter {
   // Retrieve and decrypt ciphertext using provider and condition expression
   public async retrieveAndDecrypt(
     provider: ethers.providers.Provider,
-    signer: ethers.Signer,
-    conditionExpr: ConditionExpression,
-    ciphertext: Ciphertext
+    thresholdMessageKit: ThresholdMessageKit,
+    signer?: ethers.Signer
   ): Promise<Uint8Array> {
-    const acp = await this.makeAcp(provider, signer, conditionExpr, ciphertext);
-
     const decryptionShares = await this.retrieve(
       provider,
-      conditionExpr,
-      ciphertext,
-      acp,
+      thresholdMessageKit,
       signer
     );
-
     const sharedSecret = combineDecryptionSharesSimple(decryptionShares);
-    return decryptWithSharedSecret(
-      ciphertext,
-      conditionExpr.asAad(),
-      sharedSecret
-    );
-  }
-
-  private async makeAcp(
-    provider: ethers.providers.Provider,
-    signer: ethers.Signer,
-    conditionExpr: ConditionExpression,
-    ciphertext: Ciphertext
-  ) {
-    const dkgRitual = await DkgClient.getExistingRitual(
-      provider,
-      this.ritualId
-    );
-    const authData = new AuthenticatedData(
-      dkgRitual.dkgPublicKey,
-      conditionExpr.toWASMConditions()
-    );
-
-    const headerHash = keccak256(ciphertext.header.toBytes());
-    const authorization = await signer.signMessage(headerHash);
-
-    return new AccessControlPolicy(authData, toBytes(authorization));
+    return thresholdMessageKit.decryptWithSharedSecret(sharedSecret);
   }
 
   // Retrieve decryption shares
   public async retrieve(
     provider: ethers.providers.Provider,
-    conditionExpr: ConditionExpression,
-    ciphertext: Ciphertext,
-    acp: AccessControlPolicy,
+    thresholdMessageKit: ThresholdMessageKit,
     signer?: ethers.Signer
   ): Promise<DecryptionShareSimple[]> {
     const dkgParticipants = await DkgCoordinatorAgent.getParticipants(
       provider,
       this.ritualId
     );
-    const contextStr = await conditionExpr
-      .buildContext(provider, {}, signer)
-      .toJson();
-    const { sharedSecrets, encryptedRequests } = this.makeDecryptionRequests(
-      this.ritualId,
-      ciphertext,
-      contextStr,
-      dkgParticipants,
-      acp
-    );
+    const conditionExpr = thresholdMessageKit.acp.conditions
+      ? ConditionExpression.fromWASMConditions(
+          thresholdMessageKit.acp.conditions
+        )
+      : ConditionExpression.fromObj({
+          version: ConditionExpression.VERSION,
+          condition: {},
+        });
```

Comment:
We need to somehow handle the case where `thresholdMessageKit.acp.conditions` is missing

Reply:
It should never have non-defined conditions, right?

Conditions should always be defined - that's the change here - https://github.com/nucypher/nucypher-core/pull/80

Reply:
Ok, I'm going to keep a work-around in there until we merge & release that change in `nucypher-core`.

Reply:
👍 that's fair - but let's track it so it doesn't get lost in the shuffle.

Reply:
Tracked here: https://github.com/nucypher/nucypher-ts/issues/293